### ISSUE TITLE
In which case will the debug cell button in notebook be hidden?

### ISSUE THREAD

**Issue by DonJayamanne** [microsoft/vscode-jupyter#15119]
### Discussed in https://github.com/microsoft/vscode-jupyter/discussions/15118

<div type='discussions-op-text'>

<sup>Originally posted by **realDuang** February  2, 2024</sup>
I'm trying to develop the debug feature for synapse remote server. Now I need to connect it with  vscode web notebook supported by jupyter extension.
But I found a weird thing, with the same server and config, the notebook's debug button was sometimes hidden. I want to know what scenario this is based on?
![image](https://github.com/microsoft/vscode-jupyter/assets/17807197/f9063acb-f93f-4847-98d3-e1efed995a23)
sometimes it's disappear
![image](https://github.com/microsoft/vscode-jupyter/assets/17807197/7a1e9a0b-c219-4871-af5e-3a3bde06cf4c)

I was thinking that this button was controlled via metadata in request kernelspecs, but you denied that. Then I'm confusing with it.

Below I have attached the jupyter log, I don't see any problem as it is exactly the same as when the debug cell button appears.

Visual Studio Code - Insiders (1.87.0-insider, undefined, vscode.dev)
Jupyter Extension Version: 2024.1.2024012502.
Python Extension Version: 2024.0.0.
Pylance Extension Version: 2023.12.1.
No workspace folder opened.
17:23:52.551 [error] Unable to determine the caller of the extension API for trace stack. Error: 
	at EI.determineExtensionFromCallStack (https://ms-toolsai.vscode-unpkg.net/ms-toolsai/jupyter/2024.1.2024012502+web/extension/dist/extension.web.bundle.js#vscode-extension:107:51815)
	at Object.createJupyterServerCollection (https://ms-toolsai.vscode-unpkg.net/ms-toolsai/jupyter/2024.1.2024012502+web/extension/dist/extension.web.bundle.js#vscode-extension:107:31497)
	at JupyterServerRegister.activate (webpack-internal:///./src/platform/services/jupyterServer/jupyterServerRegister.ts:60:52)
17:23:52.553 [info] Start refreshing Kernel Picker (1706865832553)
17:23:52.678 [debug] Disposing Jupyter Lab Helper
17:23:52.678 [debug] SessionManager - dispose contents manager
17:23:52.678 [debug] ShutdownSessionAndConnection - dispose session manager
17:23:52.679 [debug] Finished disposing Jupyter Lab Helper
17:23:52.679 [debug] UniversalRemoteKernelFinder: Writing 1 remote kernel connection metadata to cache
17:23:52.686 [error] Failed to cache the remote kernels. ReferenceError: Buffer is not defined
17:23:52.688 [info] End refreshing Kernel Picker (1706865832553)
17:24:00.023 [debug] Preferred Remote kernel for /A8F6376E-193C-4729-B2E9-570B631961F5/b771c1ab-1696-4f34-aac1-fe4c1b81d925/bef3408a-8d26-413b-9c5d-956cbae04d1e/vscode test.ipynb is 70321371-f435-49cb-a2c5-98cef21d44f2
17:24:00.543 [debug] No controller, hence notebook communications cannot be initialized for editor /A8F6376E-193C-4729-B2E9-570B631961F5/b771c1ab-1696-4f34-aac1-fe4c1b81d925/bef3408a-8d26-413b-9c5d-956cbae04d1e/vscode test.ipynb
17:25:59.986 [info] Handle Execution of Cells 0 for /A8F6376E-193C-4729-B2E9-570B631961F5/b771c1ab-1696-4f34-aac1-fe4c1b81d925/bef3408a-8d26-413b-9c5d-956cbae04d1e/vscode test.ipynb
17:25:59.988 [debug] start the kernel, options.disableUI=false for /A8F6376E-193C-4729-B2E9-570B631961F5/b771c1ab-1696-4f34-aac1-fe4c1b81d925/bef3408a-8d26-413b-9c5d-956cbae04d1e/vscode test.ipynb
17:25:59.990 [info] Starting Kernel startUsingRemoteKernelSpec, 5d48100f1b0130876ce64753acc7647b066f7425360b5615c8319bbfa1d49680..synapse_pyspark./python3./.-m#synapsekernel.synapse_pyspark  for '/A8F6376E-193C-4729-B2E9-570B631961F5/b771c1ab-1696-4f34-aac1-fe4c1b81d925/bef3408a-8d26-413b-9c5d-956cbae04d1e/vscode test.ipynb' (disableUI=false)
17:25:59.997 [debug] KernelProvider switched kernel to id = 5d48100f1b0130876ce64753acc7647b066f7425360b5615c8319bbfa1d49680..synapse_pyspark./python3./.-m#synapsekernel.synapse_pyspark
17:25:59.998 [debug] Initialize notebook communications for editor /A8F6376E-193C-4729-B2E9-570B631961F5/b771c1ab-1696-4f34-aac1-fe4c1b81d925/bef3408a-8d26-413b-9c5d-956cbae04d1e/vscode test.ipynb
17:25:59.998 [debug] Resolving notebook UI Comms (resolve) for /A8F6376E-193C-4729-B2E9-570B631961F5/b771c1ab-1696-4f34-aac1-fe4c1b81d925/bef3408a-8d26-413b-9c5d-956cbae04d1e/vscode test.ipynb
17:25:59.998 [debug] initialize CommonMessageCoordinator
17:25:59.999 [debug] Registering commtarget jupyter.widget
17:25:59.999 [debug] IPyWidgetMessageDispatcher.initialize
17:25:59.999 [debug] Attempting to determine version of IPyWidgets
17:26:00.000 [debug] Controller selection change completed
17:26:00.000 [debug] Waiting for IPyWidgets version
17:26:00.000 [debug] Waiting for IPyWidgets version promise
17:26:00.012 [debug] createNewKernelSession 5d48100f1b0130876ce64753acc7647b066f7425360b5615c8319bbfa1d49680..synapse_pyspark./python3./.-m#synapsekernel.synapse_pyspark
17:26:01.375 [info] https://synapse-remote-jupyter-server/: Kernel started: fdad01ef-58b8-41d0-8348-aac999da5361
17:26:01.376 [debug] Waiting for idle on (kernel): fdad01ef-58b8-41d0-8348-aac999da5361 -> unknown
17:26:02.607 [debug] Got status idle in waitForIdleOnSession
17:26:02.608 [debug] Finished waiting for idle on (kernel): fdad01ef-58b8-41d0-8348-aac999da5361 -> idle
17:26:02.608 [info] Started session for kernel startUsingRemoteKernelSpec:5d48100f1b0130876ce64753acc7647b066f7425360b5615c8319bbfa1d49680..synapse_pyspark./python3./.-m#synapsekernel.synapse_pyspark
17:26:02.609 [debug] Started running kernel initialization for /A8F6376E-193C-4729-B2E9-570B631961F5/b771c1ab-1696-4f34-aac1-fe4c1b81d925/bef3408a-8d26-413b-9c5d-956cbae04d1e/vscode test.ipynb
17:26:02.610 [debug] Executing silently Code (idle) = try:\nimport ipywidgets as _VSCODE_ipywidgets\nprint("e976ee50-99ed-4aba-9b6b-9dcd5634d07d:IPy
17:26:03.086 [warn] Unhandled message found: kernel_info_reply
17:26:23.412 [debug] Executing silently Code (completed) = try:\nimport ipywidgets as _VSCODE_ipywidgets\nprint("e976ee50-99ed-4aba-9b6b-9dcd5634d07d:IPy with 2 output(s)
17:26:23.414 [debug] Determined IPyWidgets Version as 8
17:26:23.415 [debug] Version of IPyWidgets 8 determined after 23.416s
17:26:23.415 [debug] Executing silently Code (idle) = __vsc_ipynb_file__ = "/A8F6376E-193C-4729-B2E9-570B631961F5/b771c1ab-1696-4f34-aac1-fe4c1b81d925/bef
17:26:24.866 [debug] Executing silently Code (completed) = __vsc_ipynb_file__ = "/A8F6376E-193C-4729-B2E9-570B631961F5/b771c1ab-1696-4f34-aac1-fe4c1b81d925/bef with 1 output(s)
17:26:24.867 [debug] Requesting Kernel info
17:26:24.868 [debug] Got Kernel info
17:26:24.869 [debug] End running kernel initialization, now waiting for idle
17:26:24.870 [debug] Waiting for idle on (kernel): fdad01ef-58b8-41d0-8348-aac999da5361 -> idle
17:26:24.870 [debug] Finished waiting for idle on (kernel): fdad01ef-58b8-41d0-8348-aac999da5361 -> idle
17:26:24.871 [debug] End running kernel initialization, session is idle
17:26:24.871 [debug] Updating preferred kernel for remote notebook fdad01ef-58b8-41d0-8348-aac999da5361
17:26:24.874 [info] Registering Kernel Completion Provider from kernel Synapse PySpark for language python
17:26:24.874 [info] Start refreshing Kernel Picker (1706865984874)
17:26:24.876 [debug] IPyWidgetScriptSource.initialize
17:26:24.876 [debug] IPyWidgetMessageDispatcher.initialize
17:26:24.881 [debug] Storing Preferred remote kernel for /A8F6376E-193C-4729-B2E9-570B631961F5/b771c1ab-1696-4f34-aac1-fe4c1b81d925/bef3408a-8d26-413b-9c5d-956cbae04d1e/vscode test.ipynb is fdad01ef-58b8-41d0-8348-aac999da5361
17:26:24.887 [debug] Cell Index:0 sent to kernel
17:26:24.931 [debug] IPyWidgetMessageDispatcher.initialize
17:26:24.931 [debug] Web view is ready to receive widget messages
17:26:24.931 [warn] Failed to fetch script sources ReferenceError: Buffer is not defined
17:26:24.933 [debug] Widget Message: Received IPyWidgets_kernelOptions in ScriptManager
17:26:24.939 [debug] IPyWidgetMessageDispatcher.initialize
17:26:24.940 [debug] IPyWidgetMessageDispatcher.initialize
(data is the same) [object Object]
17:26:29.257 [info] End cell 0 execution after 3.816s, completed @ 1706865989257, started @ 1706865985441
17:26:29.258 [debug] Cell 0 executed successfully
17:26:29.873 [debug] Start cell 0 execution @ undefined
17:26:29.874 [info] End cell 0 execution after 1706865989.257s, completed @ 1706865989257, started @ undefined
17:33:25.248 [debug] Disposing Jupyter Lab Helper
17:33:25.249 [debug] SessionManager - dispose contents manager
17:33:25.249 [debug] ShutdownSessionAndConnection - dispose session manager
17:33:25.265 [debug] Finished disposing Jupyter Lab Helper
17:33:25.267 [debug] UniversalRemoteKernelFinder: Writing 1 remote kernel connection metadata to cache
17:33:25.426 [error] Failed to cache the remote kernels. ReferenceError: Buffer is not defined
17:33:25.426 [info] End refreshing Kernel Picker (1706865832553)
17:33:25.426 [info] End refreshing Kernel Picker (1706865984874)
17:33:28.895 [info] Disposing request as the cell (-1) was deleted /A8F6376E-193C-4729-B2E9-570B631961F5/b771c1ab-1696-4f34-aac1-fe4c1b81d925/bef3408a-8d26-413b-9c5d-956cbae04d1e/vscode test.ipynb</div>

**Comment by realDuang** [comment]
In fact I am not very sure whether it is a bug, I just want to know under what circumstances the notebook debug function will be hidden.

If it should show up under any circumstances like you said, then we need to find out why it doesn't show the debug button.

**Comment by DonJayamanne** [comment]
@realDuang Are you still blocked by this in any way?
I'm closing this for now, please ping if you are still blocked on this or feel free to create a new issue.
